### ISSUE TITLE
Batch Point Get doesn't use converted datum

### ISSUE THREAD

**Issue by Ryan-Git** [pingcap/tidb#54746]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
CREATE TABLE `t` (`tenant_id` bigint(20) NOT NULL DEFAULT '0',`order_id` bigint(20) NOT NULL DEFAULT '0',UNIQUE KEY `uk_ten_ord` (`order_id`, `tenant_id`))  PARTITION BY HASH(`tenant_id`) PARTITIONS 32;

INSERT INTO t(tenant_id, order_id) VALUES (123, 456);

DELETE FROM t where (tenant_id, order_id) in (('123','456'));
```

### 2. What did you expect to see? (Required)
affected rows 1

### 3. What did you see instead (Required)
affected rows 0

### 4. What is your TiDB version? (Required)
up to 7.5.0, master the same I think

in https://github.com/pingcap/tidb/pull/26713, cases of ValueExpr and ParamMarkerExpr are handled(use converted int datum instead of origin string one), but the RowExpr is [left unchanged,](https://github.com/pingcap/tidb/blob/eab8700a53e2f2b2064d545ca566e804db7876df/pkg/planner/core/point_get_plan.go#L1102C13-L1102C20) which causes this problem.

Is it intentional?

**Comment by Defined2014** [comment]
It's been fixed by https://github.com/pingcap/tidb/pull/49161 and only affects v7.5.x and previous version.

**Comment by Ryan-Git** [comment]
> It's been fixed by #49161 and only affects v7.5.x and previous version.

how?  the value still uses innerX.Datum instead of dval on main. https://github.com/mjonss/tidb/blame/cc616f9ba19d9c8a9dc31ae9b95fe54077069454/pkg/planner/core/point_get_plan.go#L1124

**Comment by YangKeao** [comment]
> > It's been fixed by [#49161](https://github.com/pingcap/tidb/pull/49161) and only affects v7.5.x and previous version.
> 
> u mean plan cache re-evaluates it? It seems the value still uses innerX.Datum instead of dval on main. https://github.com/mjonss/tidb/blame/cc616f9ba19d9c8a9dc31ae9b95fe54077069454/pkg/planner/core/point_get_plan.go#L1124

@Ryan-Git You are right 👍 . It indeed exists in master branch. Ref https://github.com/pingcap/tidb/issues/60523

**Comment by Ryan-Git** [comment]
> [@Ryan-Git](https://github.com/Ryan-Git) You are right 👍 . It indeed exists in master branch. Ref [#60523](https://github.com/pingcap/tidb/issues/60523)

ok, thanks!
